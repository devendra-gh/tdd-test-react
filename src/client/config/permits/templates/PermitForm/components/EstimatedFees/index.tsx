import React from 'react';
import { IVariables } from '@tamm/app-composer';
import Table from '@tamm/ui-lib-v2-table';
import Form from '@tamm/ui-lib-v2-form';
import '../../PermitForm.less';
import Total from '@tamm/ui-lib-v2-total';
import { calculateSum } from 'client/config/permits/utils/common';
import ServiceTemplate from '@tamm/ui-lib-v2-service-template';

const EstimatedFees = (props: IVariables) => {
  const { inputGroups, i18n } = props;
  const inputGroup = inputGroups[0];
  let items = inputGroup.calculateBaseEstimatedPrice(inputGroup.values);
  if (inputGroup.values.perUnitFees !== undefined) {
    const variableItems: [] = inputGroup.calculateEstimatedFees(
      inputGroup.values,
    );
    if (
      variableItems !== undefined &&
      variableItems !== null &&
      variableItems.length !== 0
    )
      items = items.concat(variableItems);
  }

  if (items === undefined || items === null || items.length === 0) {
    items = [{ id: '0', title: 'global.permitFees', amount: 0 }];
  }
  const getFeesLabels = (feesDetails: IVariables) => {
    const { suffix, title, prefix } = feesDetails;
    if (suffix !== null && suffix !== undefined) {
      return `${i18n(title)} ${i18n(suffix)}`;
    }
    if (prefix !== null && prefix !== undefined) {
      return `${i18n(`global.${prefix.toLowerCase()}`)} ${i18n(title)}`;
    }
    return i18n(title);
  };
  return (
    <div className="marginT40">
      <Form.Fieldset
        className="marginB0"
        title={inputGroup.name}
        key={`${inputGroup.groupKey}.${inputGroup.stateKey}`}
      >
        <ServiceTemplate description={i18n('estimatedFees.description')} />
        <Table
          i18n={i18n}
          columns={inputGroup.columns.map((column: IVariables) => ({
            ...column,
            title: i18n(column.title),
          }))}
          headerHidden={false}
          items={items.map((item: IVariables) => ({
            ...item,
            title: getFeesLabels(item),
            amount: `${i18n('global.aed')} ${parseFloat(item.amount).toFixed(
              2,
            )}`,
          }))}
          selectable={false}
        />
        <div className="marginT20" />
        <div className="summary__group marginB0">
          <Total
            isValueFirst={false}
            unit={i18n('global.aed')}
            value={calculateSum(items, 'amount')}
          />
        </div>
      </Form.Fieldset>
    </div>
  );
};

export default EstimatedFees;
