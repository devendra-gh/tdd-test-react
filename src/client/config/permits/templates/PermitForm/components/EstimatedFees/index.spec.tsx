import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';
// import steps from 'client/config/renew-licence/steps';
import EstimatedFees from './index';

jest.mock('@tamm/app-composer', () => ({
  withTemplateHooks: (component: any) => component,
}));

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('permits/templates/PermitForm/components', () => {
  let props: any;

  beforeEach(() => {
    props = {
      i18n: jest.fn(),
      inputGroups: [
        {
          name: 'Table Title',
          columns: [{ title: 'Test', id: 'title' }],
          items: [{ title: 'Test', id: 'title' }],
          calculateBaseEstimatedPrice: jest.fn(),
          calculateEstimatedFees: jest.fn(),
          values: {},
        },
      ],
    };
  });

  it('should render with all props ', () => {
    const component = shallow(<EstimatedFees {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should render with all props perUnitFees defined ', () => {
    props.inputGroups[0] = {
      ...props.inputGroups[0],
      values: { perUnitFees: 10.0, baseFees: 200, baseFeesText: '' },
    };
    const component = shallow(<EstimatedFees {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should render with all props for else condition and null calculateBaseEstimatedPrice & calculateEstimatedFees', () => {
    props.currentStep = '';
    props.inputGroups[0] = {
      ...props.inputGroups[0],
      calculateBaseEstimatedPrice: () => {
        return null;
      },
      calculateEstimatedFees: () => {
        return null;
      },
    };
    const component = shallow(<EstimatedFees {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should render with all props for else condition and empty calculateBaseEstimatedPrice & calculateEstimatedFees', () => {
    props.currentStep = '';
    props.inputGroups[0] = {
      ...props.inputGroups[0],
      calculateBaseEstimatedPrice: () => {
        return [];
      },
    };
    const component = shallow(<EstimatedFees {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should render with all props for suffix', () => {
    props.currentStep = '';
    props.inputGroups[0] = {
      ...props.inputGroups[0],
      values: { perUnitFees: 10.0, baseFees: 200, baseFeesText: '' },
      calculateBaseEstimatedPrice: () => {
        return [{ title: 'Test Title', amount: 10, suffix: 'test' }];
      },
      calculateEstimatedFees: () => {
        return [{ title: 'Test Title', amount: 10, suffix: 'test' }];
      },
    };
    const component = shallow(<EstimatedFees {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should render with all props for prefix', () => {
    props.currentStep = '';
    props.inputGroups[0] = {
      ...props.inputGroups[0],
      values: { perUnitFees: 10.0, baseFees: 200, baseFeesText: '' },
      calculateBaseEstimatedPrice: () => {
        return [{ title: 'Test Title', amount: 10, prefix: 'test' }];
      },
      calculateEstimatedFees: () => {
        return [{ title: 'Test Title', amount: 10, prefix: 'test' }];
      },
    };
    const component = shallow(<EstimatedFees {...props} />);
    expect(component).toMatchSnapshot();
  });
});
