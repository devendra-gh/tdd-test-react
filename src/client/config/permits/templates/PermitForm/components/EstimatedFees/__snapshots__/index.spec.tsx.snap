// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`permits/templates/PermitForm/components should render with all props  1`] = `
<div
  className="marginT40"
>
  <Form.Fieldset
    buttonGroup={false}
    className="marginB0"
    gapSize="default"
    key="undefined.undefined"
    title="Table Title"
    twoColumns={false}
  >
    <ServiceTemplate
      description=""
      helpfulBlock={null}
      process={null}
      startLogin={null}
      tables={null}
      title=""
    />
    <LocalizedComponent(Table)
      clickable={false}
      columns={
        Array [
          Object {
            "id": "title",
            "title": undefined,
          },
        ]
      }
      commonSelection="unchecked"
      currPage={null}
      disabledSelectionVisible={false}
      editable={false}
      filter={null}
      headerHidden={false}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "estimatedFees.description",
            ],
            Array [
              "Test",
            ],
            Array [
              "global.permitFees",
            ],
            Array [
              "global.aed",
            ],
            Array [
              "global.aed",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      isSingleSelect={false}
      items={
        Array [
          Object {
            "amount": "undefined 0.00",
            "id": "0",
            "title": undefined,
          },
        ]
      }
      multiActions={null}
      onActionClick={[Function]}
      onClick={[Function]}
      onFilter={[Function]}
      onPageResize={[Function]}
      onPageTurn={[Function]}
      onRefreshClick={[Function]}
      onSearch={[Function]}
      onSelectionChange={[Function]}
      onSort={[Function]}
      onToggle={[Function]}
      pageResizeOptions={Array []}
      pageSize={null}
      rowVerticalAlign="top"
      search=""
      searchable={false}
      selectable={false}
      singleActions={null}
      size="default"
      sort={null}
      status=""
      title=""
      total={null}
    />
    <div
      className="marginT20"
    />
    <div
      className="summary__group marginB0"
    >
      <LocalizedComponent(Total)
        buttonDisabled={false}
        i18n={[Function]}
        isButtonVisible={false}
        isValueFirst={false}
        onClick={null}
        unit=""
        value={0}
      />
    </div>
  </Form.Fieldset>
</div>
`;

exports[`permits/templates/PermitForm/components should render with all props for else condition and empty calculateBaseEstimatedPrice & calculateEstimatedFees 1`] = `
<div
  className="marginT40"
>
  <Form.Fieldset
    buttonGroup={false}
    className="marginB0"
    gapSize="default"
    key="undefined.undefined"
    title="Table Title"
    twoColumns={false}
  >
    <ServiceTemplate
      description=""
      helpfulBlock={null}
      process={null}
      startLogin={null}
      tables={null}
      title=""
    />
    <LocalizedComponent(Table)
      clickable={false}
      columns={
        Array [
          Object {
            "id": "title",
            "title": undefined,
          },
        ]
      }
      commonSelection="unchecked"
      currPage={null}
      disabledSelectionVisible={false}
      editable={false}
      filter={null}
      headerHidden={false}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "estimatedFees.description",
            ],
            Array [
              "Test",
            ],
            Array [
              "global.permitFees",
            ],
            Array [
              "global.aed",
            ],
            Array [
              "global.aed",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      isSingleSelect={false}
      items={
        Array [
          Object {
            "amount": "undefined 0.00",
            "id": "0",
            "title": undefined,
          },
        ]
      }
      multiActions={null}
      onActionClick={[Function]}
      onClick={[Function]}
      onFilter={[Function]}
      onPageResize={[Function]}
      onPageTurn={[Function]}
      onRefreshClick={[Function]}
      onSearch={[Function]}
      onSelectionChange={[Function]}
      onSort={[Function]}
      onToggle={[Function]}
      pageResizeOptions={Array []}
      pageSize={null}
      rowVerticalAlign="top"
      search=""
      searchable={false}
      selectable={false}
      singleActions={null}
      size="default"
      sort={null}
      status=""
      title=""
      total={null}
    />
    <div
      className="marginT20"
    />
    <div
      className="summary__group marginB0"
    >
      <LocalizedComponent(Total)
        buttonDisabled={false}
        i18n={[Function]}
        isButtonVisible={false}
        isValueFirst={false}
        onClick={null}
        unit=""
        value={0}
      />
    </div>
  </Form.Fieldset>
</div>
`;

exports[`permits/templates/PermitForm/components should render with all props for else condition and null calculateBaseEstimatedPrice & calculateEstimatedFees 1`] = `
<div
  className="marginT40"
>
  <Form.Fieldset
    buttonGroup={false}
    className="marginB0"
    gapSize="default"
    key="undefined.undefined"
    title="Table Title"
    twoColumns={false}
  >
    <ServiceTemplate
      description=""
      helpfulBlock={null}
      process={null}
      startLogin={null}
      tables={null}
      title=""
    />
    <LocalizedComponent(Table)
      clickable={false}
      columns={
        Array [
          Object {
            "id": "title",
            "title": undefined,
          },
        ]
      }
      commonSelection="unchecked"
      currPage={null}
      disabledSelectionVisible={false}
      editable={false}
      filter={null}
      headerHidden={false}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "estimatedFees.description",
            ],
            Array [
              "Test",
            ],
            Array [
              "global.permitFees",
            ],
            Array [
              "global.aed",
            ],
            Array [
              "global.aed",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      isSingleSelect={false}
      items={
        Array [
          Object {
            "amount": "undefined 0.00",
            "id": "0",
            "title": undefined,
          },
        ]
      }
      multiActions={null}
      onActionClick={[Function]}
      onClick={[Function]}
      onFilter={[Function]}
      onPageResize={[Function]}
      onPageTurn={[Function]}
      onRefreshClick={[Function]}
      onSearch={[Function]}
      onSelectionChange={[Function]}
      onSort={[Function]}
      onToggle={[Function]}
      pageResizeOptions={Array []}
      pageSize={null}
      rowVerticalAlign="top"
      search=""
      searchable={false}
      selectable={false}
      singleActions={null}
      size="default"
      sort={null}
      status=""
      title=""
      total={null}
    />
    <div
      className="marginT20"
    />
    <div
      className="summary__group marginB0"
    >
      <LocalizedComponent(Total)
        buttonDisabled={false}
        i18n={[Function]}
        isButtonVisible={false}
        isValueFirst={false}
        onClick={null}
        unit=""
        value={0}
      />
    </div>
  </Form.Fieldset>
</div>
`;

exports[`permits/templates/PermitForm/components should render with all props for prefix 1`] = `
<div
  className="marginT40"
>
  <Form.Fieldset
    buttonGroup={false}
    className="marginB0"
    gapSize="default"
    key="undefined.undefined"
    title="Table Title"
    twoColumns={false}
  >
    <ServiceTemplate
      description=""
      helpfulBlock={null}
      process={null}
      startLogin={null}
      tables={null}
      title=""
    />
    <LocalizedComponent(Table)
      clickable={false}
      columns={
        Array [
          Object {
            "id": "title",
            "title": undefined,
          },
        ]
      }
      commonSelection="unchecked"
      currPage={null}
      disabledSelectionVisible={false}
      editable={false}
      filter={null}
      headerHidden={false}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "estimatedFees.description",
            ],
            Array [
              "Test",
            ],
            Array [
              "global.test",
            ],
            Array [
              "Test Title",
            ],
            Array [
              "global.aed",
            ],
            Array [
              "global.test",
            ],
            Array [
              "Test Title",
            ],
            Array [
              "global.aed",
            ],
            Array [
              "global.aed",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      isSingleSelect={false}
      items={
        Array [
          Object {
            "amount": "undefined 10.00",
            "prefix": "test",
            "title": "undefined undefined",
          },
          Object {
            "amount": "undefined 10.00",
            "prefix": "test",
            "title": "undefined undefined",
          },
        ]
      }
      multiActions={null}
      onActionClick={[Function]}
      onClick={[Function]}
      onFilter={[Function]}
      onPageResize={[Function]}
      onPageTurn={[Function]}
      onRefreshClick={[Function]}
      onSearch={[Function]}
      onSelectionChange={[Function]}
      onSort={[Function]}
      onToggle={[Function]}
      pageResizeOptions={Array []}
      pageSize={null}
      rowVerticalAlign="top"
      search=""
      searchable={false}
      selectable={false}
      singleActions={null}
      size="default"
      sort={null}
      status=""
      title=""
      total={null}
    />
    <div
      className="marginT20"
    />
    <div
      className="summary__group marginB0"
    >
      <LocalizedComponent(Total)
        buttonDisabled={false}
        i18n={[Function]}
        isButtonVisible={false}
        isValueFirst={false}
        onClick={null}
        unit=""
        value={20}
      />
    </div>
  </Form.Fieldset>
</div>
`;

exports[`permits/templates/PermitForm/components should render with all props for suffix 1`] = `
<div
  className="marginT40"
>
  <Form.Fieldset
    buttonGroup={false}
    className="marginB0"
    gapSize="default"
    key="undefined.undefined"
    title="Table Title"
    twoColumns={false}
  >
    <ServiceTemplate
      description=""
      helpfulBlock={null}
      process={null}
      startLogin={null}
      tables={null}
      title=""
    />
    <LocalizedComponent(Table)
      clickable={false}
      columns={
        Array [
          Object {
            "id": "title",
            "title": undefined,
          },
        ]
      }
      commonSelection="unchecked"
      currPage={null}
      disabledSelectionVisible={false}
      editable={false}
      filter={null}
      headerHidden={false}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "estimatedFees.description",
            ],
            Array [
              "Test",
            ],
            Array [
              "Test Title",
            ],
            Array [
              "test",
            ],
            Array [
              "global.aed",
            ],
            Array [
              "Test Title",
            ],
            Array [
              "test",
            ],
            Array [
              "global.aed",
            ],
            Array [
              "global.aed",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      isSingleSelect={false}
      items={
        Array [
          Object {
            "amount": "undefined 10.00",
            "suffix": "test",
            "title": "undefined undefined",
          },
          Object {
            "amount": "undefined 10.00",
            "suffix": "test",
            "title": "undefined undefined",
          },
        ]
      }
      multiActions={null}
      onActionClick={[Function]}
      onClick={[Function]}
      onFilter={[Function]}
      onPageResize={[Function]}
      onPageTurn={[Function]}
      onRefreshClick={[Function]}
      onSearch={[Function]}
      onSelectionChange={[Function]}
      onSort={[Function]}
      onToggle={[Function]}
      pageResizeOptions={Array []}
      pageSize={null}
      rowVerticalAlign="top"
      search=""
      searchable={false}
      selectable={false}
      singleActions={null}
      size="default"
      sort={null}
      status=""
      title=""
      total={null}
    />
    <div
      className="marginT20"
    />
    <div
      className="summary__group marginB0"
    >
      <LocalizedComponent(Total)
        buttonDisabled={false}
        i18n={[Function]}
        isButtonVisible={false}
        isValueFirst={false}
        onClick={null}
        unit=""
        value={20}
      />
    </div>
  </Form.Fieldset>
</div>
`;

exports[`permits/templates/PermitForm/components should render with all props perUnitFees defined  1`] = `
<div
  className="marginT40"
>
  <Form.Fieldset
    buttonGroup={false}
    className="marginB0"
    gapSize="default"
    key="undefined.undefined"
    title="Table Title"
    twoColumns={false}
  >
    <ServiceTemplate
      description=""
      helpfulBlock={null}
      process={null}
      startLogin={null}
      tables={null}
      title=""
    />
    <LocalizedComponent(Table)
      clickable={false}
      columns={
        Array [
          Object {
            "id": "title",
            "title": undefined,
          },
        ]
      }
      commonSelection="unchecked"
      currPage={null}
      disabledSelectionVisible={false}
      editable={false}
      filter={null}
      headerHidden={false}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "estimatedFees.description",
            ],
            Array [
              "Test",
            ],
            Array [
              "global.permitFees",
            ],
            Array [
              "global.aed",
            ],
            Array [
              "global.aed",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      isSingleSelect={false}
      items={
        Array [
          Object {
            "amount": "undefined 0.00",
            "id": "0",
            "title": undefined,
          },
        ]
      }
      multiActions={null}
      onActionClick={[Function]}
      onClick={[Function]}
      onFilter={[Function]}
      onPageResize={[Function]}
      onPageTurn={[Function]}
      onRefreshClick={[Function]}
      onSearch={[Function]}
      onSelectionChange={[Function]}
      onSort={[Function]}
      onToggle={[Function]}
      pageResizeOptions={Array []}
      pageSize={null}
      rowVerticalAlign="top"
      search=""
      searchable={false}
      selectable={false}
      singleActions={null}
      size="default"
      sort={null}
      status=""
      title=""
      total={null}
    />
    <div
      className="marginT20"
    />
    <div
      className="summary__group marginB0"
    >
      <LocalizedComponent(Total)
        buttonDisabled={false}
        i18n={[Function]}
        isButtonVisible={false}
        isValueFirst={false}
        onClick={null}
        unit=""
        value={0}
      />
    </div>
  </Form.Fieldset>
</div>
`;
